src/protocol/basics/reveal.rs

```diff
@@ -1,24 +1,24 @@
 use std::iter::{repeat, zip};
 
-use crate::ff::Field;
+use crate::ff::{ArithmeticOps, Field};
 use crate::protocol::context::{Context, MaliciousContext, SemiHonestContext};
-use crate::secret_sharing::{MaliciousReplicated, Replicated, SecretSharing};
+use crate::secret_sharing::{MaliciousReplicated, Replicated, SecretSharing, SharedValue};
 use crate::{error::Error, helpers::Direction, protocol::RecordId};
 use async_trait::async_trait;
 use embed_doc_image::embed_doc_image;
 use futures::future::{try_join, try_join_all};
 
 /// Trait for reveal protocol to open a shared secret to all helpers inside the MPC ring.
 #[async_trait]
-pub trait Reveal<F: Field> {
+pub trait Reveal<V: SharedValue + ArithmeticOps> {
```

Comment:
I'm seeing a few of these, which is fine, but I wonder if we might define a simple alias trait like `trait ArithmeticShare: SharedValue + ArithmeticOps`.